src/unitxt/operators.py

```diff
@@ -508,75 +508,121 @@ def process_value(self, value: Any) -> Any:
+        "": 30,
+    }
+    remove_existing_whitespaces: Optional[bool] = False
 
     def verify(self):
         assert (
-            isinstance(self.suffixes, list) or isinstance(self.suffixes, dict)
-        ), f"Argument 'suffixes' should be either a list or a dictionary, whereas it is of type {type(self.suffixes)}"
-
-        if isinstance(self.suffixes, dict):
-            for k, v in self.suffixes.items():
-                assert isinstance(
-                    k, str
-                ), f"suffixes should map strings, whereas key {k!s} is of type {type(k)}"
-                assert isinstance(
-                    v, int
-                ), f"suffixes should map to ints, whereas value {v!s} is of type {type(v)}"
-        else:
-            for k in self.suffixes:
-                assert isinstance(
-                    k, str
-                ), f"suffixes should be a list of strings, whereas member {k!s} is of type {type(k)}"
+            self.prefixes or self.suffixes
+        ), "At least one of prefixes/suffixes should be not None."
+        for arg, arg_name in zip(
+            [self.prefixes, self.suffixes], ["prefixes", "suffixes"]
+        ):
+            assert (
+                arg is None or isinstance(arg, list) or isinstance(arg, dict)
+            ), f"Argument {arg_name} should be either None or a list or a dictionary, whereas it is of type {type(arg)}"
+
+            if arg is None:
+                continue
+            if isinstance(arg, dict):
+                for k, v in arg.items():
+                    assert isinstance(
+                        k, str
+                    ), f"{arg_name} should map strings, whereas key {k!s} is of type {type(k)}"
+                    assert isinstance(
+                        v, int
+                    ), f"{arg_name} should map to ints, whereas value {v!s} is of type {type(v)}"
+            else:
+                for k in arg:
+                    assert isinstance(
+                        k, str
+                    ), f"{arg_name} should be a list of strings, whereas member {k!s} is of type {type(k)}"
         super().verify()
 
     def prepare(self):
-        self.pats = (
-            self.suffixes
-            if isinstance(self.suffixes, list)
-            else [k for k, v in self.suffixes.items()]
-        )
-        total_weight = (
-            len(self.pats)
-            if isinstance(self.suffixes, list)
-            else sum([v for k, v in self.suffixes.items()])
-        )
-        self.weights = (
-            [1.0 / total_weight] * len(self.pats)
-            if isinstance(self.suffixes, list)
-            else [float(self.suffixes[p]) / total_weight for p in self.pats]
-        )
+        # Being an artifact, prepare is invoked before verify. Here we need verify before the actions
+        self.verify()
+        self.pats = {"prefixes": None, "suffixes": None}
+        self.weights = {"prefixes": None, "suffixes": None}
+
+        for end, end_name in zip(
```

Comment:
I think it would be better to use functions for this

(self.prefix_patterns, self.prefix_probabilities) = calculate_distributions(self.prefixes)
(self.suffix_patterns, self.suffix_probabilities) = calculate_distributions(self.suffixes)

Reply:
Done. Indeed cleaner. Thanks!